backend/python/app/services/implementations/entity_service.py

```diff
@@ -0,0 +1,51 @@
+from ...models.entity import Entity
+from ...models import db
+from ..interfaces.entity_service import IEntityService
+
+
+class EntityService(IEntityService):
+    def __init__(self, logger):
+        self.logger = logger
+
+    def get_entities(self):
+        # Entity is a SQLAlchemy model, we can use convenient methods provided
+        # by SQLAlchemy like query.all() to query the data
+        return [result.to_dict() for result in Entity.query.all()]
+
+    def get_entity(self, id):
+        # get queries by the primary key, which is id for the Entity table
+        entity = Entity.query.get(id)
+        if entity is None:
+            raise Exception("Invalid id")
```

Comment:
This would be a place to use the logger in addition to raising the exception, `self.logger.error("error message here")`